### ISSUE TITLE
Manage Crate Owners / Add Owner UX is confusing

### ISSUE THREAD

**Issue by golddranks** [rust-lang/crates.io#2164]
Hi, I just used the Manage Crate Owners / Add Owner functionality in crates.io for the first time. I found the UX of the form not to be quite ideal:

* It doesn't detect if the user I'm trying to pass ownership exists at all:
<img width="940" alt="スクリーンショット 2020-01-27 22 13 41" src="https://user-images.githubusercontent.com/2675542/73177408-4ccf5000-4152-11ea-88c6-ab49f432e383.png">

* Because of this, I can't be entirely sure whether I typed the username correctly.
  * I have a nagging feeling, after typoing once, that I might be passing the ownership to a malicious fake user.
  * After page reload, it doesn't show that an invitation has been sent, and there is no way to revoke an invitation in the case I typoed.

While not maybe a priority number one of crates.io, I'd appreciate if these concerns were heard.

**Comment by Zexbe** [comment]
Yeah, I just sent an invite to the wrong person on accident, and I don't have a way to cancel it.
